File: DristorApp/Data/Models/Product.cs
using System;
namespace DristorApp.Data.Models
{
    public class Product
    {
        public int Id { set; get; }
        public string Name { set; get; }

        public virtual ICollection<Ingredient> Ingredients { set; get; }
        public virtual ICollection<ProductVariant> ProductVariants { set; get; }

    }
}

File: frontend/src/app/models/product.ts
import {IIngredient} from "./ingredient";
import { IProductVariant } from "./product-variant";

export interface IProduct {
  id?: number;
  name: string;

  ingredients?: IIngredient[];
  productVariants?: IProductVariant[];
}

export interface IProductCreate {
  name: string;

  ingredients: number[];
  productVariants: number[];
}

export interface IProductUpdate extends IProductCreate { }
